arrow-optics-test/src/main/kotlin/arrow/test/generators/Generators.kt

```diff
@@ -1,123 +1,7 @@
-
-fun <A, B, C, D, E, F, G> Gen.Companion.tuple7(genA: Gen<A>, genB: Gen<B>, genC: Gen<C>, genD: Gen<D>, genE: Gen<E>, genF: Gen<F>, genG: Gen<G>): Gen<Tuple7<A, B, C, D, E, F, G>> =
-  Gen.bind(genA, genB, genC, genD, genE, genF, genG) { a: A, b: B, c: C, d: D, e: E, f: F, g: G -> Tuple7(a, b, c, d, e, f, g) }
-
-fun <A, B, C, D, E, F, G, H> Gen.Companion.tuple8(genA: Gen<A>, genB: Gen<B>, genC: Gen<C>, genD: Gen<D>, genE: Gen<E>, genF: Gen<F>, genG: Gen<G>, genH: Gen<H>): Gen<Tuple8<A, B, C, D, E, F, G, H>> =
-  Gen.bind(Gen.tuple7(genA, genB, genC, genD, genE, genF, genG), genH) { tuple: Tuple7<A, B, C, D, E, F, G>, h: H -> Tuple8(tuple.a, tuple.b, tuple.c, tuple.d, tuple.e, tuple.f, tuple.g, h) }
-
-fun <A, B, C, D, E, F, G, H, I> Gen.Companion.tuple9(genA: Gen<A>, genB: Gen<B>, genC: Gen<C>, genD: Gen<D>, genE: Gen<E>, genF: Gen<F>, genG: Gen<G>, genH: Gen<H>, genI: Gen<I>): Gen<Tuple9<A, B, C, D, E, F, G, H, I>> =
-  Gen.bind(Gen.tuple8(genA, genB, genC, genD, genE, genF, genG, genH), genI) { tuple: Tuple8<A, B, C, D, E, F, G, H>, i: I -> Tuple9(tuple.a, tuple.b, tuple.c, tuple.d, tuple.e, tuple.f, tuple.g, tuple.h, i) }
-
-fun <A, B, C, D, E, F, G, H, I, J> Gen.Companion.tuple10(genA: Gen<A>, genB: Gen<B>, genC: Gen<C>, genD: Gen<D>, genE: Gen<E>, genF: Gen<F>, genG: Gen<G>, genH: Gen<H>, genI: Gen<I>, genJ: Gen<J>): Gen<Tuple10<A, B, C, D, E, F, G, H, I, J>> =
-  Gen.bind(Gen.tuple9(genA, genB, genC, genD, genE, genF, genG, genH, genI), genJ) { tuple: Tuple9<A, B, C, D, E, F, G, H, I>, j: J -> Tuple10(tuple.a, tuple.b, tuple.c, tuple.d, tuple.e, tuple.f, tuple.g, tuple.h, tuple.i, j) }
-
-fun <B> Gen.Companion.option(gen: Gen<B>): Gen<Option<B>> =
-  gen.orNull().map { it.toOption() }
-
-fun <E, A> Gen.Companion.either(genE: Gen<E>, genA: Gen<A>): Gen<Either<E, A>> {
-  val genLeft = genE.map<Either<E, A>> { Left(it) }
-  val genRight = genA.map<Either<E, A>> { Right(it) }
-  return Gen.oneOf(genLeft, genRight)
-}
-
-fun <E, A> Gen.Companion.validated(genE: Gen<E>, genA: Gen<A>): Gen<Validated<E, A>> =
-  Gen.either(genE, genA).map { Validated.fromEither(it) }
-
-fun <A> Gen.Companion.`try`(genA: Gen<A>, genThrowable: Gen<Throwable> = throwable()): Gen<Try<A>> =
-  Gen.either(genThrowable, genA).map { it.fold({ Failure(it) }, { Success(it) }) }
-
-fun <A> Gen.Companion.nonEmptyList(gen: Gen<A>): Gen<NonEmptyList<A>> =
-  gen.flatMap { head -> Gen.list(gen).map { NonEmptyList(head, it) } }
-
-fun <K : Comparable<K>, V> Gen.Companion.sortedMapK(genK: Gen<K>, genV: Gen<V>): Gen<SortedMapK<K, V>> =
-  Gen.bind(genK, genV) { k: K, v: V -> sortedMapOf(k to v) }.map { it.k() }
-
-fun <K, V> Gen.Companion.mapK(genK: Gen<K>, genV: Gen<V>): Gen<MapK<K, V>> =
-  Gen.map(genK, genV).map { it.k() }
-
-fun <A> Gen.Companion.listK(genA: Gen<A>): Gen<ListK<A>> = Gen.list(genA).map { it.k() }
-
-fun <A> Gen.Companion.sequenceK(genA: Gen<A>): Gen<SequenceK<A>> = Gen.list(genA).map { it.asSequence().k() }
-
+// TODO move to Arrow Core Test
```

Comment:
I was reviewing this case with @danimontoya and there is something to take into account. Not all these extensions of `interface Gen<T>`  are included in the JAR. If the current `build/libs/arrow-optics-test-0.10.5-SNAPSHOT.jar` is inspected, it doesn't include:
* char
* throwable
* fatalThrowable
* intSmall
All of them are not defined in terms of generic types and they are only found in the same Gradle project. However, they aren't included in the JAR to be used for other projects.

Reply:
None of these should be defined in `arrow-optics-test`. All of these should be defined in `arrow-core-test`. `throwable` and `fatalThrowable` could also live in Arrow Fx.

Reply:
Right, however, the current definition of `char` it doesn't make it possible. It's not included in the JAR so it cannot be shared with other projects.

Reply:
Okay, I will temporarily add it here under the `arrow-optics-test` package.